### PR TITLE
Update oracle test container version to 18.4.0-full

### PR DESCRIPTION
**Pull request by chenjian2664**
## Description
Fix https://github.com/trinodb/trino/issues/5901

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by chenjian2664** [comment]
> Could you run stress tests? Those scripts were introduced to avoid flaky tests.

Thank you! I do not have full context about this scripts.
The main thing I want to do this to allow build longer objects names for Oracle tests. In what condition we can remove the scripts?(like run 10 times, failure 1 or more details requirements). So, we can keep the scripts is total ok for me.

**Comment by chenjian2664** [comment]
Only update oracle container version at current

**Comment by ebyhr** [comment]
> Fix https://github.com/trinodb/trino/issues/5901

There're many TODO comments about this issue. We shouldn't close it in this PR.

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java [hunk 1]
```diff
@@ -234,8 +234,7 @@ public void testDeleteAllDataFromTable()
     public void testRowLevelDelete()
     {
         skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_ROW_LEVEL_DELETE));
-        // TODO (https://github.com/trinodb/trino/issues/5901) Use longer table name once Oracle version is updated
-        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_row_delete", "AS SELECT * FROM region")) {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_row_level_delete_" + randomNameSuffix(), "AS SELECT * FROM region")) {
```
Comment (ebyhr) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:237]: `TestTable` adds suffix internally.  Please remove `randomNameSuffix`.

Comment (chenjian2664) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:237]: thanks for your reminder!

**Review by hashhar — CHANGES_REQUESTED** [review]
If we want we should add new test with newer version since 11.2 is still the oldest supported version by Oracle and we want to make sure that we can find and fix bugs against it.

So I don't think this PR would allow to remove the TODOs anyway since 11.2 still needs to be tested. Sometime after 11.2 is out of support (Dec 2023) we can revisit this PR. For now I think we shouldn't merge this.

**Comment by chenjian2664** [comment]
@hashhar Make sense. How about: 

1. How about fix the TODO in the base test cases and only make the different in the Oracle connector test class
2. For the new version, we could postpone it

**Comment by ebyhr** [comment]
> How about fix the TODO in the base test cases and only make the different in the Oracle connector test class

Do you mean renaming tables in `BaseConnectorTest` and override each test method in `BaseOracleConnectorTest`? I disagree if so. It's hard to keep in sync. We may forget to update the Oracle test when changing the base test.

**Comment by chenjian2664** [comment]
> > How about fix the TODO in the base test cases and only make the different in the Oracle connector test class
> 
> Do you mean renaming tables in `BaseConnectorTest` and override each test method in `BaseOracleConnectorTest`? 

Yes.
> I disagree if so. It's hard to keep in sync. We may forget to update the Oracle test when changing the base test.

We could only override the naming not all the test

**Comment by ebyhr** [comment]
> We could only override the naming not all the test

That's possible, but overriding the naming only for the purpose doesn't make sense. It reduces readability and it doesn't bring any benefit. I would recommend closing this PR now.

**Comment by chenjian2664** [comment]
> > We could only override the naming not all the test
> 
> That's possible, but overriding the naming only for the purpose doesn't make sense. It reduces readability and it doesn't bring any benefit. I would recommend closing this PR now.

Can. I meet this when want to open the merge feature for the Oracle, there are exist tests cases can not fit Oracle naming constraints, for these cases we shouldn't shorter existing name I think, do you agree?

**Comment by ebyhr** [comment]
> for these cases we shouldn't shorter existing name

Why?

**Comment by chenjian2664** [comment]
> > for these cases we shouldn't shorter existing name
> 
> Why?

It's reducing the readability for the existing test, and the adjust only for the Oracle.

**Comment by mahic** [comment]
@hashhar Would it make sense to revisit this as 11 is out of support now. Trino is thus running tests on an unsupported version of Oracle. It would be a nice side-effect if merging in a newer version of the test container would result in [supporting the latest JDBC driver version](https://github.com/trinodb/trino/pull/22984)

**Comment by hashhar** [comment]
@mahic Thanks for revisiting this. Yes, I think this makes sense to revive now. Do you already plan on sending a PR?

**Comment by chenjian2664** [comment]
@hashhar @ebyhr Do you think it's worth to reopen it? I am still facing the naming issues while modifying the Oracle tests

**Pull request closed without merging**